### DISCUSSION TITLE
@types/leaflet - event type not correctly inferred

### DISCUSSION THREAD

**Discussion by lajerca** [DefinitelyTyped/DefinitelyTyped#60721] [Issues with a @types package]
I think the event type is not correctly inferred because this line should be at the bottom of the list:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4a254ca8d58c398690998cf914d708f5acc15627/types/leaflet/index.d.ts#L370

for example in the following code `e` is a generic event which doesn't contain `latlng` because the handler function's type is `LeafletEventHandlerFn` instead of `LeafletMouseEventHandlerFn`. 

```javascript
  map.on("click", (e) =>
    const coords = e.latlng; // TYPE ERROR
  });
  ```

Since for typescript the type of `"click"` is a string, the first typed method takes precedence over the others, and the type check never makes it to the correct type listed, which for clarity should be this:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4a254ca8d58c398690998cf914d708f5acc15627/types/leaflet/index.d.ts#L390-L392

Tested the fix locally and it works, therefore I suggest moving the generic type to the bottom of the list.

[a comment by an automated account is not shown]

**Comment by lajerca** [comment]
while this gets fixed for anybody having the same issue the easiest workaround is to use the more explicit 
```javascript
 map.on({
    click: (e) => {
      const coords = e.latlng;
    }
  });
  ```

**Comment by MysteryBlokHed** [comment]
I just opened #60724 to change this
